appcli/configuration_manager.py

```diff
@@ -59,6 +88,49 @@ def __init__(self, cli_context: CliContext, configuration: Configuration):
         self.cli_configuration: Configuration = configuration
         self.variables_manager: VariablesManager = cli_context.get_variables_manager()
 
+    def validate_configuration(self):
+        """Validates all settings files that have associated schema files."""
+
+        # Define all the config directories and files.
+        settings_schema: Path = self.cli_context.get_app_configuration_file_schema()
+        stack_settings_schema: Path = (
+            self.cli_context.get_stack_configuration_file_schema()
+        )
+        overrides_dir: Path = self.cli_context.get_baseline_template_overrides_dir()
+        templates_dir: Path = self.cli_context.get_configurable_templates_dir()
```

Comment:
The files from overrides/templates get copied into the generated config dir. We don't want to copy the schema files into that dir, otherwise programs might think that they are part of the configuration (i.e. Teraflow will try and process all json/yaml files it sees in the definition directories).

So we probably need to ensure they don't get copied over. We probably want a way to have an exception to the rule. I.e. some system might actually want `schema.json` files to go across. So we need a way to allow for that. To be honest that's probably another ticket, I don't see an immediate use case. Maybe it can be as simple as appending `.appcli` to the file, and we strip the suffix on copy.

Reply:
Will be handled in APPCLI-114